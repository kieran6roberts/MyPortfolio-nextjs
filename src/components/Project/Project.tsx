import Image from "next/image";
import Link from "next/link";
import { BiMouse } from "react-icons/bi";
import { VscChevronRight } from "react-icons/vsc";
import { motion as m } from "framer-motion";

import Button from "../Button/Button";
import SubHeader from "../SubHeader/SubHeader";
import List from "../List/List";

export type ProjectProps = {
    title: string,
    description: string,
    siteLink?: string,
    githubLink?: string,
    captions?: string
    image: string,
    stackImages: {
        __typename: string,
        fileName: string
    }[],
    __typename?: string,
    stackNames: string[]
};

export default function Project({ 
    siteLink, 
    githubLink, 
    description, 
    title, 
    image, 
    captions, 
    stackImages, 
    stackNames }: ProjectProps): React.ReactElement {

    return (
        <div className="mb-32 md:px-20 lg:px-32 md:mx-12 bg-light overflow-hidden shadow">
            <div className="bg-light px-4">
                <SubHeader title={title}/>
                <Link href={`projects/${title}`}>
                    <m.div className="group relative flex justify-center w-max-40 xl:w-max m-auto p-1 mb-8 cursor-pointer"
                        whileHover={{ scale: 1.05, rotate: 2 }}
                        whileTap={{ rotate: 45, translateY: -200 }}
                        transition={{ duration: 0.5 }}>
                        <div className="absolute top-0 right-0 left-0 bottom-0 bg-purple-400 bg-opacity-90 z-10 transition duration-300 ease-in group-hover:bg-opacity-20"/>
                        <BiMouse className="absolute z-20 h-12 w-12 top-2/4 -mt-6 text-md text-light transition duration-150 ease-in group-hover:text-transparent animate-pulse" />
                        <Image src={`/images/${image}`}
                        alt={captions}
                        width={775}
                        height={350}
                        objectFit="cover"/>
                    </m.div>
                </Link>
                <p className="text-xs italic text-dark text-center mb-8">
                    {captions}
                </p>
                <div className="flex flex-col md:flex-row py-8">
                    <Button link={siteLink}
                    color="bg-light text-dark mb-2 md:mb-0 md:mr-4">
                        visit website
                    </Button>
                    <Button link={githubLink}
                    color="bg-offLight text-dark">
                        github repo
                    </Button>
                </div>
                <div className="bg-gradient-to-b pl-1 from-purple-400 via-purple-600 to-yellow-400">
                    <h4 className="bg-light text-md text-sec uppercase my-8 xxl:my-24 pl-4">
                        project description
                    </h4>
                </div>
                <p className="text-xs text-dark py-2">
                    {description}
                </p>
                <div className="bg-gradient-to-b pl-1 from-purple-400 via-purple-600 to-yellow-400">
                    <h4 className="bg-light text-md text-sec uppercase my-8 xxl:my-24 pl-4">
                        tech stack
                    </h4>
                </div>
                <List images={stackImages} names={stackNames} />
                <div className="w-max h-max my-16 bg-gradient-to-r from-purple-400 to-purple-600 p-0.5">
                    <Link href={`/projects/${title}`}>
                        <a className={`group flex items-center bg-pri w-max flex align-items whitespace-nowrap text-xxs text-light font-bold uppercase py-2 px-3 cursor-pointer transition duration-100 ease-in hover:bg-light hover:text-dark`}>
                            <span className="block pr-4">
                                case-study
                            </span>
                            <VscChevronRight className="text-xs transition duration-50 ease-in opacity-0 group-hover:opacity-100"/>
                        </a>
                    </Link>
                </div>
            </div>
        </div>
    )
};