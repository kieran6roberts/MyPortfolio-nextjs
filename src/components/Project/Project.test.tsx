import * as React from "react";
import { render, RenderResult } from "@testing-library/react";

import Project from "@/components/Project/Project";

let documentBody: RenderResult;

describe("<Hero />", () => {
    const testProps = {
            title: "turncoat brewery",
            description: "description of turncoat",
            siteLink: "turncoat.com",
            githubLink: "github-turncoat.com",
            captions: ["turncoat caption"],
            image: "turncoat-image.webp",
            stackImages: [{ __typename: "turnocat typename", fileName: "stack.png" }],
            stackNames: ["html"],
            images: [{ fileName: "img1" } , { fileName: "img2" }, { fileName: "img3" }, { fileName: "img4" }]
    };

    beforeEach(() => {
        documentBody = render(<Project {...testProps}/>);
    })
    test("renders", () => {
        expect(documentBody.getByText("description of turncoat")).toBeInTheDocument();
    });
});