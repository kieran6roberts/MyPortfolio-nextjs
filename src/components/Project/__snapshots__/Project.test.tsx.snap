// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> renders snapshot 1`] = `
<article
  className="mb-32 overflow-hidden shadow md:px-4 lg:px-12 md:mx-6 bg-light"
>
  <div
    className="px-4 bg-light"
  >
    <header
      className="relative w-screen text-left"
    >
      <h2
        className="text-gray-200 text-xxxl w-offScreen whitespace-nowrap"
      >
        turncoat brewery
      </h2>
      <h3
        className="absolute px-4 font-bold bg-white border-l-4 shadow-md bg-opacity-30 top-1/4 text-md md:text-lg border-pri"
        id="projects"
      >
        turncoat brewery
      </h3>
    </header>
    <div
      className="relative w-full max-w-md mx-auto mb-24 2xl:mb-56 md:max-w-7xl h-hero 2xl:h-lgContain"
    >
      <div
        className="absolute z-10 w-2/4 h-full right-6 sm:-right-16 lg:-right-32 2xl:-right-56 -bottom-12 2xl:h-full"
      >
        <div
          className="bg-black border-4 border-black shadow-2xl max-h-mobileH h-5/6 rounded-xl w-max"
        >
          <div
            style={
              Object {
                "boxSizing": "border-box",
                "display": "inline-block",
                "margin": 0,
                "maxWidth": "100%",
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            <div
              style={
                Object {
                  "boxSizing": "border-box",
                  "display": "block",
                  "maxWidth": "100%",
                }
              }
            >
              <img
                alt=""
                aria-hidden={true}
                role="presentation"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjAwIiBoZWlnaHQ9IjQwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                style={
                  Object {
                    "display": "block",
                    "maxWidth": "100%",
                  }
                }
              />
            </div>
            <img
              alt="cap3"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": "contain",
                  "objectPosition": "offTop",
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "visibility": "hidden",
                  "width": 0,
                }
              }
            />
          </div>
        </div>
      </div>
      <div
        className="absolute left-0 w-full shadow-md md:w-4/6 md:left-24 2xl:w-full 2xl:-left-56 bottom-1/2 h-3/6"
      >
        <div
          className="w-full h-full"
        >
          <div
            style={
              Object {
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "left": 0,
                "margin": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <img
              alt="cap1"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": "cover",
                  "objectPosition": "left top",
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "visibility": "hidden",
                  "width": 0,
                }
              }
            />
          </div>
        </div>
      </div>
      <div
        className="absolute left-0 w-5/6 shadow-md md:left-24 md:w-4/6 2xl:-right-1/2 2xl:w-full -bottom-4 h-1/2"
      >
        <div
          className="w-full h-full"
        >
          <div
            style={
              Object {
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "left": 0,
                "margin": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <img
              alt="cap2"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": "cover",
                  "objectPosition": "left top",
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "visibility": "hidden",
                  "width": 0,
                }
              }
            />
          </div>
        </div>
        <div
          className="absolute invisible hidden overflow-hidden bg-black border-4 border-black shadow-2xl bottom-4 2xl:block 2xl:visible max-h-mobileH h-5/6 rounded-xl w-max"
        >
          <div
            style={
              Object {
                "boxSizing": "border-box",
                "display": "inline-block",
                "margin": 0,
                "maxWidth": "100%",
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            <div
              style={
                Object {
                  "boxSizing": "border-box",
                  "display": "block",
                  "maxWidth": "100%",
                }
              }
            >
              <img
                alt=""
                aria-hidden={true}
                role="presentation"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjAwIiBoZWlnaHQ9IjM4MCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                style={
                  Object {
                    "display": "block",
                    "maxWidth": "100%",
                  }
                }
              />
            </div>
            <img
              alt="cap4"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": "contain",
                  "objectPosition": "offTop",
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "visibility": "hidden",
                  "width": 0,
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="flex flex-col pb-8 2xl:justify-center sm:flex-row 2xl:pb-32"
    >
      <a
        className="group bg-light mb-2 2xl:mx-8 sm:mb-0 sm:mr-4 w-44 2xl:w-80 flex align-items whitespace-nowrap text-xxs text-dark font-bold uppercase py-2 px-3 2xl:py-6 2xl:px-6 border-2 border-pri cursor-pointer transition duration-50 ease-in hover:bg-pri focus:outline-none focus:ring-4 focus:ring-yellow-400 active:bg-sec"
        href="turncoat.com"
        rel="noopener"
        target="_blank"
      >
        <span
          className="block pr-4 group-hover:text-light "
        >
          visit website
        </span>
        <svg
          className="text-xs transition ease-in text-pri duration-50 group-hover:text-white"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M1.5 13A1.5 1.5 0 003 14.5h8a1.5 1.5 0 001.5-1.5V9a.5.5 0 00-1 0v4a.5.5 0 01-.5.5H3a.5.5 0 01-.5-.5V5a.5.5 0 01.5-.5h4a.5.5 0 000-1H3A1.5 1.5 0 001.5 5v8zm7-11a.5.5 0 01.5-.5h5a.5.5 0 01.5.5v5a.5.5 0 01-1 0V2.5H9a.5.5 0 01-.5-.5z"
            fillRule="evenodd"
          />
          <path
            clipRule="evenodd"
            d="M14.354 1.646a.5.5 0 010 .708l-8 8a.5.5 0 01-.708-.708l8-8a.5.5 0 01.708 0z"
            fillRule="evenodd"
          />
        </svg>
      </a>
      <a
        className="group bg-offLight 2xl:mx-8 w-44 2xl:w-80 flex align-items whitespace-nowrap text-xxs text-dark font-bold uppercase py-2 px-3 2xl:py-6 2xl:px-6 border-2 border-pri cursor-pointer transition duration-50 ease-in hover:bg-pri focus:outline-none focus:ring-4 focus:ring-yellow-400 active:bg-sec"
        href="github-turncoat.com"
        rel="noopener"
        target="_blank"
      >
        <span
          className="block pr-4 group-hover:text-light "
        >
          github repo
        </span>
        <svg
          className="text-xs transition ease-in text-pri duration-50 group-hover:text-white"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M1.5 13A1.5 1.5 0 003 14.5h8a1.5 1.5 0 001.5-1.5V9a.5.5 0 00-1 0v4a.5.5 0 01-.5.5H3a.5.5 0 01-.5-.5V5a.5.5 0 01.5-.5h4a.5.5 0 000-1H3A1.5 1.5 0 001.5 5v8zm7-11a.5.5 0 01.5-.5h5a.5.5 0 01.5.5v5a.5.5 0 01-1 0V2.5H9a.5.5 0 01-.5-.5z"
            fillRule="evenodd"
          />
          <path
            clipRule="evenodd"
            d="M14.354 1.646a.5.5 0 010 .708l-8 8a.5.5 0 01-.708-.708l8-8a.5.5 0 01.708 0z"
            fillRule="evenodd"
          />
        </svg>
      </a>
    </div>
    <div
      className="pl-1 bg-gradient-to-b from-purple-400 via-purple-600 to-blue-800"
    >
      <h4
        className="pl-4 my-8 uppercase bg-light text-md text-sec 2xl:my-24"
      >
        project description
      </h4>
    </div>
    <article
      className="text-xs"
    >
      <p>
        description of turncoat
      </p>
    </article>
    <div
      className="pl-1 bg-gradient-to-b from-purple-400 via-purple-600 to-blue-800"
    >
      <h4
        className="pl-4 my-8 uppercase bg-light text-md text-sec 2xl:my-24"
      >
        tech stack
      </h4>
    </div>
    <ul
      className="flex flex-col flex-wrap py-2 lg:flex-row"
    >
      <li
        className="flex items-center justify-start my-4 lg:pb-12 lg:my-0 lg:mr-12 lg:flex-col lg:justify-center"
      >
        <div
          className="block mr-4 lg:mr-0"
        >
          <div
            style={
              Object {
                "boxSizing": "border-box",
                "display": "inline-block",
                "margin": 0,
                "maxWidth": "100%",
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            <div
              style={
                Object {
                  "boxSizing": "border-box",
                  "display": "block",
                  "maxWidth": "100%",
                }
              }
            >
              <img
                alt=""
                aria-hidden={true}
                role="presentation"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iODAiIGhlaWdodD0iNDAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                style={
                  Object {
                    "display": "block",
                    "maxWidth": "100%",
                  }
                }
              />
            </div>
            <img
              alt="[object Object] logo"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "visibility": "hidden",
                  "width": 0,
                }
              }
            />
          </div>
        </div>
        <span
          className="block ml-2 text-xs font-bold lg:ml-0 whitespace-nowrap text-dark"
        >
          html
        </span>
      </li>
    </ul>
    <div
      className="my-16 2xl:my-32 py-0.5 px-2"
    >
      <a
        className="group w-56 flex items-center justify-center bg-pri border-pri border-2 flex align-items whitespace-nowrap text-xxs text-light font-bold uppercase py-2 px-4 2xl:py-4 2xl:px-6 cursor-pointer transition duration-100 ease-in hover:bg-white hover:text-dark focus:outline-none focus:ring-4 focus:ring-yellow-400"
        href="/projects/turncoat%20brewery"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <span
          className="block pr-4"
        >
          case-study
        </span>
        <svg
          aria-label="top of the page"
          className="text-xs transition ease-in transform translate-x-0 text-light duration-50 group-hover:translate-x-8 group-hover:text-sec "
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M10.072 8.024L5.715 3.667l.618-.62L11 7.716v.618L6.333 13l-.618-.619 4.357-4.357z"
            fillRule="evenodd"
          />
        </svg>
      </a>
    </div>
  </div>
</article>
`;
